### PR TITLE
ab: modify course options

### PR DESCRIPTION
**Pull request by annagav**
When modifying existing assignment, shows a list of courses (formset) with an extra row, where you can create a new course or change exiting ones.

### CODE REVIEW

File: instructor/common.py [hunk 1]
```diff
@@ -188,19 +187,24 @@ def course_modify(request):
                 return redirect("common_assignments_edit_strains")
         else:
             # change this course's name or code
-            form = CourseForm(request.POST, instance=assignment.course)
-            if form.is_valid():
-                form.save()
+            formset = CourseFormSet(request.POST)
+            formset.clean()
```
Comment (giocalitri) [inline instructor/common.py]: question: what does `formset.clean()` do?

File: instructor/common.py [hunk 2]
```diff
@@ -188,19 +187,24 @@ def course_modify(request):
...
+    formset = CourseFormSet(queryset=models.Course.objects.all())
     all_courses = models.Course.objects.all()
-    return render_to_response('instructor/course_setup.html',
+    return render_to_response('instructor/course_modify.html',
                               {'courses': all_courses,
-                               'form': form,
-                               'new':  request.session['new']},
+                               'formset': formset,
+                               },
                               context_instance=RequestContext(request))
```
Comment (giocalitri) [inline instructor/common.py:232]: Need to indent better

File: instructor/templates/instructor/course_modify.html [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                }
+                var url = '/ab/assignments/course_modify/';
+                $.ajax({
+                    url: url,
+                    type: 'POST',
+                    data: data
+                });
+            });
+        });
+
```
Comment (giocalitri) [inline instructor/templates/instructor/course_modify.html:5]: Question: is there a good reason to put this javascript inline in the template?

**Comment by annagav** [comment]
Optimized the code a bit.

File: instructor/common.py (hunk 1 above)
Comment (annagav) [inline instructor/common.py]: You are right. This is not needed.

File: instructor/templates/instructor/course_modify.html (hunk 3 above)
Comment (annagav) [inline instructor/templates/instructor/course_modify.html:5]: Just for convenience. I actually think I can rewrite this functionality in much better way.

File: instructor/common.py [hunk 4]
```diff
@@ -188,19 +186,24 @@ def course_modify(request):
...
-    return render_to_response('instructor/course_setup.html',
-                              {'courses': all_courses,
-                               'form': form,
-                               'new':  request.session['new']},
+    return render_to_response('instructor/course_modify.html',
+                              {
+                                  'courses': all_courses,
+                                  'formset': formset,
+                              },
                               context_instance=RequestContext(request))
```
Comment (giocalitri) [inline instructor/common.py:232]: nit: the right indentation is:

```
return render_to_response(
    'instructor/course_modify.html',
    {
        'courses': all_courses,
        'formset': formset,
    },
    context_instance=RequestContext(request)
)
```

**Pull request merged**